hls-test-utils/src/Test/Hls/Util.hs

```diff
@@ -384,3 +386,10 @@ getCompletionByLabel desiredLabel compls =
         Nothing -> liftIO . assertFailure $
             "Completion with label " <> show desiredLabel
             <> " not found in " <> show (fmap (^. L.label) compls)
+
+-- ---------------------------------------------------------------------
+-- Run with a canonicalized temp dir
+withTempDir :: (FilePath -> IO a) -> IO a
+withTempDir f = System.IO.Extra.withTempDir $ \dir -> do
+  dir' <- canonicalizePath dir
+  f dir'
```

Comment:
Because I need run pragma insertion tests in an isolated directory without `hie.yaml` explicitly since its behavior is not the same between ghc-9.2 and others(ghc-9.2 enabled `GADTSyntax` implicitly).

https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0380-ghc2021.rst#proposed-change

Reply:
I mean, why do we need canonicalization?

Reply:
It's from ghcide's test. https://github.com/haskell/haskell-language-server/blob/b1bf5499155e259341e4868863b0fd743b6bd65f/ghcide/test/exe/Main.hs#L6975-L6981

Reply:
Oh... because of macOS. I think it's probably helpful to refer that comment or add that comment.